storybookjs/storybook
# fix typing on useArgs

Issue: `useArgs` is typed incorrectly.

## What I did
Fixed `useArgs` to take a type parameter, `P`, which should be the type of the args for your story. By default, it's set to `Args` so that it's backwards compatible. Additionally, fixed the `resetArgs` function to be a `(string[]) => void` rather than a `([string]) => void`, as a `[string]` makes no sense in this context (and is incorrect).

## How to test
This is just a type change, so this can be tested by ensuring that things are typed correctly in your story when using `useArgs<YourStoryArgType>`.


If your answer is yes to any of these, please make sure to include it in your PR.
